packages/std/src/math/int64.rs

```diff
@@ -0,0 +1,1047 @@
+    #[inline]
+    pub fn wrapping_add(self, other: Self) -> Self {
+        Self(self.0.wrapping_add(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    #[inline]
+    pub fn wrapping_sub(self, other: Self) -> Self {
+        Self(self.0.wrapping_sub(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    #[inline]
+    pub fn wrapping_mul(self, other: Self) -> Self {
+        Self(self.0.wrapping_mul(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    #[inline]
+    pub fn wrapping_pow(self, other: u32) -> Self {
+        Self(self.0.wrapping_pow(other))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_add(self, other: Self) -> Self {
+        Self(self.0.saturating_add(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_sub(self, other: Self) -> Self {
+        Self(self.0.saturating_sub(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_mul(self, other: Self) -> Self {
+        Self(self.0.saturating_mul(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_pow(self, exp: u32) -> Self {
+        Self(self.0.saturating_pow(exp))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn abs_diff(self, other: Self) -> Uint64 {
+        Uint64(self.0.abs_diff(other.0))
+    }
+}
+
+impl From<u32> for Int64 {
+    fn from(val: u32) -> Self {
+        Int64(val.into())
+    }
+}
+
+impl From<u16> for Int64 {
+    fn from(val: u16) -> Self {
+        Int64(val.into())
+    }
+}
+
+impl From<u8> for Int64 {
+    fn from(val: u8) -> Self {
+        Int64(val.into())
+    }
+}
+
+impl From<i64> for Int64 {
+    fn from(val: i64) -> Self {
+        Int64(val)
+    }
+}
+
+impl From<i32> for Int64 {
+    fn from(val: i32) -> Self {
+        Int64(val.into())
+    }
+}
+
+impl From<i16> for Int64 {
+    fn from(val: i16) -> Self {
+        Int64(val.into())
+    }
+}
+
+impl From<i8> for Int64 {
+    fn from(val: i8) -> Self {
+        Int64(val.into())
+    }
+}
+
+impl TryFrom<&str> for Int64 {
+    type Error = StdError;
+
+    fn try_from(val: &str) -> Result<Self, Self::Error> {
+        Self::from_str(val)
+    }
+}
+
+impl FromStr for Int64 {
+    type Err = StdError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.parse::<i64>() {
+            Ok(u) => Ok(Self(u)),
+            Err(e) => Err(StdError::generic_err(format!("Parsing Int64: {}", e))),
+        }
+    }
+}
```

Comment:
Shouldn't this be left unimplemented, and only rely on / provide `TryFrom`? I mean, `TryFrom` is used for conversions that can fail, while `From` is (typically) for the ones that cannot.

Reply:
`From` should only be implemented if it cannot error, but `FromStr` can error.
It is also implemented for the native types (in fact we are using the `i64` impl under the hood here in that `parse::<i64>()` call.

Reply:
Ah, yes, this is `FromStr`, missed that.